File: 409-LongestPalindrome/409-LongestPalindrome.java
// Last updated: 31/07/2025, 21:16:41
class Solution {
    public int longestPalindrome(String s) {
        int[] charCounts = new int[128];
        for (char c : s.toCharArray()) {
            charCounts[c]++;
        }
        int sum = 0;
        boolean oddFreq = false;
        for (int freq : charCounts) {
            if (freq % 2 == 0) {
                sum += freq;
            } else {
                sum += (freq - 1);
                oddFreq = true;
            }
        }
        return oddFreq ? sum + 1 : sum;
    }
}

File: 409-LongestPalindrome/409-LongestPalindrome.py
# Last updated: 31/07/2025, 21:16:47
class Solution:
    def longestPalindrome(self, s: str) -> int:
        c = Counter(s)
        res = 0
        odd = False
        for i in c:
            if c[i]%2 == 0:
                res += c[i]
            else:
                res += c[i]-1
                odd = True

        return res if not odd else res+1